doc/compiler/control/CompilationControl.md

```diff
@@ -0,0 +1,304 @@
+- DEEPSTEADY_STATE
+  - Decisions focus on maintaining throughput while minimizing CPU
+    consumed by the JIT (including the Sampler Thread).
+
+## Available Resources
+
+The JIT should minimize its impact on application performance.
+Therefore, how aggressively it can compile methods depends on
+the resources available to the JVM. Multiple compilation threads may
+only be activated if the system is not resource constrained.
+To minimize memory usage, only one higher optimization level
+(hot, very-hot, scorching) compile is performed at a time.
+Additionally, the maximum amount of memory available for a compilation
+depends on the amount of physical memory available.
+
+## JIT Hooks
+
+The JIT registers several hooks with the VM. These hooks are invoked
+when the environment changes at well defined points. These changes
+include Class Loading / Unloading, Code Redefinition, Garbage
+Collection, etc. These events can cause invalidations both in the
+generated code and the compilation. The former is addressed using
+[Runtime Assumptions](../runtime/RuntimeAssumption.md); the latter
+is addressed by aborting the compilation and possibly reattempting
+it.
+
+## Shared Classes Cache
+
+The existence of the Shared Classes Cache (SCC) has a sigificant
+influence on the heuristics; this is because of SCC enables
+Ahead of Time (AOT) Compilation. See the [AOT docs](../aot) for
+technical details on the AOT feature.
+
+Because AOT compilations enable very fast startup, the JIT tries
+to maximize the number of compiled bodies it can load from the SCC
+(referred to as AOT Load). As a consequence, the invocation counts
+for AOT Compilations are lower than JIT Compilations. Additionally,
+the invocation counts for an AOT Load is even lower - by default
+it is 20.
+
+By very their nature, AOT method bodies have to be
+relocated before they can be executed. As a consequence, the
+quality of AOT compiled code is, in general, less than that of JIT
+compiled code. Therefore, the JIT uses AOT to speed up the early phase
+of the application, namely Startup, and then depends on
+recompilation to achieve peak Steady State performance. The idea
+is that the set of methods that need to be recompiled for
+peak Steady State performance is smaller than the total number of
+methods in the application. Therefore, the JIT consumes less memory
+and CPU.
+
+## JIT Server
+
+JIT Server is a feature in OpenJ9 that allows clients to offload
+compilations to a server; see the [JIT Server docs](../jitserver)
+for more details.
+
+## Special Modes
+
+OpenJ9 has a few special flags that the user can specify to indicate
+to the JIT the kind of behaviour desired. This section outlines
+these modes.
+
+### -Xtune:virtualized
+
+This mode emphasises fast Startup and Rampup, and low CPU utilization,
+at the cost of a small throughput degradation. It is designed for
+use in virtualized environments.
+
+### -Xtune:throughput
+
+This mode emphases maximizing throughput performance.
```

Comment:
@mpirvu Anything else I should add, i.e. tradeoffs, who it's designed for, etc.?

Reply:
emphases --> emphasizes

Reply:
at the expense of other performance metrics like start-up time, footprint and CPU consumption